exercises/src/main/scala/fpinscala/parallelism/Nonblocking.scala

```diff
@@ -55,30 +65,36 @@ object Nonblocking {
     def eval(es: ExecutorService)(r: => Unit): Unit =
       es.submit(new Callable[Unit] { def call = r })
 
-
     def map2[A,B,C](p: Par[A], p2: Par[B])(f: (A,B) => C): Par[C] =
       es => new Future[C] {
-        def apply(cb: C => Unit): Unit = {
+        def apply(cb: C => Unit)(kErr: Throwable => Unit): Unit = {
+          def cbOrErr(a: A, b: B) =
+            try { cb(f(a, b)) }
+            catch { case err: Throwable => kErr(err) }
```

Comment:
Generally in Scala you don't want to catch `Throwable`, you want to use `NonFatal(e)` so that you exclude `Error`s and `InterruptedException` from your catch, those you shouldn't try to recover from. Although in this case just using `NonFatal(e)` would probably not be a great idea because your latch might never get released and you can have other trouble. So in real code you would want to have a more thought out management of exception than what's suggested by the book here.